src/identities/providers/github/GitHubProvider.ts

```diff
@@ -280,102 +280,92 @@ class GitHubProvider extends Provider {
       );
     }
     providerToken = await this.checkToken(providerToken, authIdentityId);
-    let pageNum = 1;
-    while (true) {
-      const request = this.createRequest(
-        `${this.apiUrl}/user/following?per_page=100&page=${pageNum}`,
-        {
-          method: 'GET',
-        },
-        providerToken,
-      );
-      const response = await fetch(request);
-      if (!response.ok) {
-        if (response.status === 401) {
-          throw new identitiesErrors.ErrorProviderUnauthenticated(
-            `Invalid access token`,
+
+    const foundIdentityIds: Set<IdentityId> = new Set();
+    for (const identityGroup of ['following', 'followers'] as const) {
+      let cursor: string | undefined;
+      while (true) {
+        const request = this.createRequest(
+          `${this.apiUrl}/graphql`,
+          {
+            method: 'POST',
+            body: this.getConnectedIdentityDatasRequestBody(authIdentityId, identityGroup, cursor)
```

Comment:
Since it's a special request constructor, can you indicate the name of it as some sort of graphql request?

Also I need docblock comments explaining how it works, and which part of GitHub API documentation you're referring to.

Reply:
1. user Query Gets a User with an auth token: https://docs.github.com/en/graphql/reference/queries#user
2. The User object contains the FollowerConnection object: https://docs.github.com/en/graphql/reference/objects#user
3. The FollowerConnection object Contains User objects: https://docs.github.com/en/graphql/reference/objects#followerconnection
4. The User object contains the following fields:
  - login
  - name
  - email
  - name

Reply:
Fixed:
```ts
/**
   * Returns a string suitable for use as the request body to the GitHub GraphQL endpoint.
   * This is used to construct a query that returns either the `followers` or the `following` of a user.
   *
   * Schemas Used:
   * - https://docs.github.com/en/graphql/reference/queries#user
   * - https://docs.github.com/en/graphql/reference/objects#user
   * - https://docs.github.com/en/graphql/reference/objects#followerconnection
   *
   * @param authIdentityId - The GitHub authentication token to use when getting user data
   * @param identityGroup - Specify whether the GraphQL query requests the `followers` or the `following` of a user
   * @param cursor - cursor for pagination,
   * this can be retrieved from `.data.user[identityGroup].pageinfo.endCursor`
   * of the JSON body on a response for a request made with the return value of this method as the body.
   */
  protected getConnectedIdentityDatasGraphQLBody(
    authIdentityId: IdentityId,
    identityGroup: 'following' | 'followers',
    cursor?: string
  ): string {
```